### ISSUE TITLE
Query panel improvements

### ISSUE THREAD

**Issue by bharatkashyap** [mui/toolpad#2997]
Collecting tasks be taken up once #2393 is merged

### Bugs
- [x] https://github.com/mui/mui-toolpad/issues/3051

### Code

- [ ] [packages/toolpad-app/src/toolpad/AppEditor/NodeMenu.tsx](https://github.com/mui/mui-toolpad/pull/2393/files/7b6b40c6b5378c1cd73a2fddb24339f2f0532b39#diff-9d07cdb31a496f58836283180bc5f1c6e0f2ea7dfc8bf61ddcedd0310b090890)

   ```ts
   const isAction = React.useMemo(() => {
       if (latestDeletedNode?.type === 'query' && latestDeletedNode?.attributes?.mode === 'mutation') {
   ```
   We can likely abstract this in a function and reuse in the filter where we split queries into "queries" and "actions". It's probably also not necessary to memoize. The memoization call is going to be more computationally expensive than two nested property lookups and a boolean operation.

### Nomenclature
- [ ] [packages/toolpad-app/src/runtime/ToolpadApp.tsx](https://github.com/mui/mui-toolpad/pull/2393/files/7b6b40c6b5378c1cd73a2fddb24339f2f0532b39#diff-83e5ff8d1da57db0956541265e4200e0e04d8da7584e77f0caea0d1e1604f0fe)
   ```ts
      let kind: 'query' | 'action' = 'query';
   ```
   - either we call it "action" everywhere and just do a translation to "mutation", right before we read/write to a file
   - or we call it "mutation" everywhere and name it "action" just right before we display in the UI
   
  The reason is that this prevents us from mismatching and constant translation all around the codebase. Ideally if the rename is done in a large PR, you start with the latter, just to avoid introducing many unnecessary chamges to unrelated code. Then in a separate PR update the codebase to the former.

### UI/UX

- [ ] The "Run" and "Save" button locations and types seem fine to me, but design-wise it feels like they need more vertical space around them. Not sure what the best way to improve that would be though.

   <img width="769" alt="Screenshot 2023-11-23 at 16 37 41" src="https://github.com/mui/mui-toolpad/assets/10789765/3a73fc67-d5ed-4f04-a3c5-105ab1aef5dd">

- [ ] Creating a new file in a custom function doesn't let us name the file, right? And the button at the bottom does not take the full-width of the popover so that may feel a bit weird. We can maybe improve the UX here later!

   <img width="304" alt="Screenshot 2023-11-23 at 16 37 34" src="https://github.com/mui/mui-toolpad/assets/10789765/b6f66907-cbe9-436b-adbd-7773f1bc806e">
